app/controllers/old_exams_controller.rb

```diff
@@ -67,16 +54,19 @@ def new
       flash[:alert] = 'Ordner nicht gefunden.'
     end
 
-    @old_exam = OldExam.new
+    @old_exam = @old_folder.old_exams.new
     @existing_titles = OldExam.existing_titles
     @existing_examiners = OldExam.existing_examiners
   end
 
   def edit
-    @old_exam = OldExam.find(params[:id])
-    if @old_exam.has_invalid_date?
-      flash[:alert] = "Datum in Datenbank (#{date_before_type_cast}) ist fehlerhaft. Bitte das Datum korrigieren."
+    old_folder_id = params[:old_folder_id]
+
+    if old_folder_id.nil?
+      flash[:alert] = 'Kein Ordner angegeben.'
     end
+
+    @old_exam = OldExam.find(params[:id])
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Unscoped call to OldExam#find](https://www.codacy.com/app/lippertsjan/klausurtool-ror/file/2131358069/issues/source?bid=3141297&fileBranchId=3143249#l69)